streaming/src/message/message.h

```diff
@@ -16,74 +17,123 @@ enum class StreamingMessageType : uint32_t {
   MAX = Message
 };
 
+enum class StreamingBarrierType : uint32_t { GlobalBarrier = 0 };
+
+struct StreamingBarrierHeader {
+  StreamingBarrierType barrier_type;
+  uint64_t barrier_id;
+  inline bool IsGlobalBarrier() {
+    return StreamingBarrierType::GlobalBarrier == barrier_type;
+  }
+};
+
 constexpr uint32_t kMessageHeaderSize =
     sizeof(uint32_t) + sizeof(uint64_t) + sizeof(StreamingMessageType);
 
+constexpr uint32_t kBarrierHeaderSize = sizeof(StreamingBarrierType) + sizeof(uint64_t);
+
 /// All messages should be wrapped by this protocol.
 //  DataSize means length of raw data, message id is increasing from [1, +INF].
 //  MessageType will be used for barrier transporting and checkpoint.
 ///      +----------------+
-///      | DataSize=U32   |
+///      | PayloadSize=U32|
 ///      +----------------+
 ///      | MessageId=U64  |
 ///      +----------------+
 ///      | MessageType=U32|
 ///      +----------------+
-///      | Data=var       |
+///      | Payload=var    |
 ///      +----------------+
+/// Payload field contains barrier header and carried buffer if message type is
+/// global/partial barrier.
+///
+/// Barrier's Payload field:
+///      +----------------------------+
+///      | StreamingBarrierType=U32   |
+///      +----------------------------+
+///      | barrier_id=U64             |
+///      +----------------------------+
+///      | carried_buffer=var         |
+///      +----------------------------+
 
 class StreamingMessage {
  private:
-  std::shared_ptr<uint8_t> message_data_;
-  uint32_t data_size_;
+  std::shared_ptr<uint8_t> payload_data_;
```

Comment:
`payload_` and `payload_size_` will be enough?

Reply:
done.